app/components/Modals/TokenModal/utils.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+import { uniqueId } from 'lodash'
+
+const TOKEN_ID_PREFIX = '_ug'
+
+export const getNewTokenItem = (networkId: string) => ({
+  id: uniqueId(TOKEN_ID_PREFIX),
+  symbol: '',
+  scriptHash: '',
+  networkId,
+  isUserGenerated: true
+})
+
+export const validateTokens = (tokens: Array<TokenItemType>) => {
```

Comment:
I think giving the return value of this object a flowtype would have made it obvious that `InputErrorType` wasn't matching in my previous comment.  e.g.:

```js
type TokenValidationResult = {
  errorItemId: number | null,
  errorType: InputErrorType | null,
  errorMessage: string | null
}

export const validateTokens = (tokens: Array<TokenItemType>): TokenValidationResult => { ... }
```